File: chain-sdk/clojure/src/on_chain_api/specs/factory_v3_dto.clj
(ns on-chain-api.specs.factory-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def factory-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :pool_count) string?
   (ds/opt :tx_count) string?
   (ds/opt :total_volume_usd) string?
   (ds/opt :total_volume_eth) string?
   (ds/opt :total_fees_usd) string?
   (ds/opt :total_fees_eth) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :total_value_locked_usd) string?
   (ds/opt :total_value_locked_eth) string?
   (ds/opt :total_value_locked_usd_untracked) string?
   (ds/opt :total_value_locked_eth_untracked) string?
   (ds/opt :owner) string?
   (ds/opt :vid) int?
   })

(def factory-v3-dto-spec
  (ds/spec
    {:name ::factory-v3-dto
     :spec factory-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/FactoryV3DTO.js
goog.provide('API.Client.FactoryV3DTO');

/**
 * The Uniswap Factory entity is responsible for storing aggregate information across all Uniswap pairs. It can be used to view stats about total liquidity, volume, amount of pairs and more.
 * @record
 */
API.Client.FactoryV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.FactoryV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.FactoryV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.FactoryV3DTO.prototype.blockNumber;

/**
 * Factory address.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.id;

/**
 * Amount of pools created.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.poolCount;

/**
 * Amount of transactions all time.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.txCount;

/**
 * Total volume all time in derived USD.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalVolumeUsd;

/**
 * Total volume all time in derived ETH.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalVolumeEth;

/**
 * Total swap fees all time in USD.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalFeesUsd;

/**
 * All volume even through less reliable USD values.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalFeesEth;

/**
 * All volume even through less reliable USD values.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.untrackedVolumeUsd;

/**
 * Total value locked derived in USD.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalValueLockedUsd;

/**
 * Total value locked derived in ETH.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalValueLockedEth;

/**
 * Total value locked derived in USD untracked.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalValueLockedUsdUntracked;

/**
 * Total value locked derived in ETH untracked.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.totalValueLockedEthUntracked;

/**
 * Current owner of the factory.
 * @type {!string}
 * @export
 */
API.Client.FactoryV3DTO.prototype.owner;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.FactoryV3DTO.prototype.vid;

